### PR TITLE
Relax Max: Make Content-Security-Policy report-only in production

### LINKED ISSUE #1753: Default CSP to report-only for Lift 3.0.0
As a transitory step, Lift 3.0.0 should only report CSP violations,
not enforce them, even in production. A slight deviation from “secure
by default”, but it should make the transition to Lift 3 more straightforward.
Lift 3.1.0 will move us to a production default of enforced security policies.

### PR DESCRIPTION
**Pull request by Shadowfiend**
This is done so that transition to Lift 3.0.0 will be less
bumpy. See [the associated mailing list discussion](https://groups.google.com/forum/#!searchin/liftweb/CSP%7Csort:date/liftweb/2lDoWpT6ufk/166IsbPjBAAJ) for
more. I also added a spec for `SecurityRules`, although
it doesn't quite cover everything.

Fixes #1753 .

### CODE REVIEW

File: web/webkit/src/test/scala/net/liftweb/http/SecurityRulesSpec.scala [hunk 1]
```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2011 WorldWide Conferencing, LLC
```
Comment (eltimn) [inline web/webkit/src/test/scala/net/liftweb/http/SecurityRulesSpec.scala]: Should probably change this to 2016.

**Comment by eltimn** [comment]
:+1 other than the copyright year

File: web/webkit/src/test/scala/net/liftweb/http/SecurityRulesSpec.scala [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+      ContentSecurityPolicy().reportUri must_== Some(ContentSecurityPolicy.defaultReportUri)
+    }
+
+    "provide [X-]Content-Security-Policy if enforcement is enabled" in {
+      ContentSecurityPolicy()
+        .headers(enforce = true, logViolations = true)
+        .collect {
+          case (headerName, _) if headerName.contains("Content-Security-Policy") =>
+            headerName
+        } must_== List("Content-Security-Policy", "X-Content-Security-Policy")
```
Comment (farmdawgnation) [inline web/webkit/src/test/scala/net/liftweb/http/SecurityRulesSpec.scala]: It might be better to use `must contain` on the list this `collect` generates than asserting over the entire list at once. The errors in this case (when they occur) are a bit hard to parse out as a human and we don't _really_ care about the ordering for this to be valid.

**Comment by farmdawgnation** [comment]
Overall, this looks good to me. :+1:

File: web/webkit/src/test/scala/net/liftweb/http/SecurityRulesSpec.scala (hunk 2 above)
Comment (Shadowfiend) [inline web/webkit/src/test/scala/net/liftweb/http/SecurityRulesSpec.scala]: Because this is security stuff, I'm trying to be a little more specific about checking that it includes these and only these values and headers. Maybe there's a verified that checks that all contents are exactly the same but not necessarily in order (`must contain` doesn't fail if there are unexpected things, iirc). 

That said, the error messages aren't particularly hard to parse imo: you get both the result and the expectation printed out (plus that weird string diff, but that's not useful 99% of the time anyway), right?

File: web/webkit/src/test/scala/net/liftweb/http/SecurityRulesSpec.scala (hunk 1 above)
Comment (Shadowfiend) [inline web/webkit/src/test/scala/net/liftweb/http/SecurityRulesSpec.scala]: I'll try and do that. This is why I strongly dislike these headers... so many of them are out of date anyway. < grump grump grump >

**Comment by Shadowfiend** [comment]
Kk, fixed the copyright date so I think I'm just going to go ahead and send this in.
